ADR/0033-upstreaming-integration-service.md

```diff
@@ -0,0 +1,271 @@
+are associated with the app `my-application`. When triggered, it will extract the same image and
+source details that we extract today to supply to a new [TestSubject].
+
+### Release Initiation, as owned by the release-service
+
+For the purpose of this ADR, we will only say that release-service will take over the
+responsibility of initiating Releases. That will be described in more detail in a subsequent,
+forthcoming ADR.
+
+Note, in this pattern, [Release] construction looks a lot like [TestSubject] construction. The
+triggered system knows enough about the triggering system to optionally initiate itself.
+
+## Use Cases in Detail
+
+**Using integration-service as a remote API from Jenkins.** A hypothetical platform team might want
+to make use of [integration-service] from a jenkins pipeline. For example, perhaps their jenkins job
+performs CI on a git repository that contains pullspecs which should be mirrored to some environment
+of theirs. Before mirroring, they want to make sure that the same test works against dozens of
+different kubernetes configurations on different public clouds. That platform team first
+creates tekton pipelines that contains the logic to perform the test, and use [Dynamic Resource
+Allocation APIs] to request provisioning of those clusters.
+They then configure dozens of [IntegrationTestScenario] resources, one for each kubernetes
+configuration they want to target. When a new set of pullspecs is proposed to their git repo,
+Jenkins responds. Jenkins then creates a single [TestSubject] in the kubernetes namespace,
+representing the new set of images under test. [integration-service] responds, inspecting all of the
+defined [IntegrationTestScenarios] and it runs tests of the [TestSubject] against
+each. The aggregate results are stored in the `status` of the [TestSubject] for jenkins to monitor
+and report back to the platform team in CI.
+
+**Using integration-service with TestSubject automation.** A hypothetical platform team might want
+to make use of [integration-service] in conjunction with tekton pipelines that they use for
+building images. Imagine, every night they have a cron trigger which rebuilds their images using
+a static Dockerfile that installs the latest debian packages available to a base image and then
+rebuilds all of their utility images on that new base image. Their pipelines expose the pullspecs of
+all of those images as results. The platform team configures a [TestSubjectConstructor] which
+instructs the [test-subject-construction-controller] to watch all PipelineRuns in their namespace with
+a certain label. As those builds finish, the [test-subject-construction-controller] constructs new
+[TestSubjects] by combining the image list from the **control** [TestSubject] with the new pullspec
+found in the result with the name described in the [TestSubjectConstructor]. The platform team
+benefits from integration-service's [two-phase architecture]. They build reports on the history of
```

Comment:
We'll probably need to adjust the two phase architecture to account for the new paradigm. While the core idea will remain the same, I think we'll need to specify that we'd be comparing the new `TestSubject` with the newest **control** `TestSubject`, and detecting if any of the other component images (ones that were not part of the `TestSubject` update were changed.
This would probably only work for `TestSubjects` which were constructed with a `TestSubjectConstructor` if I'm not mistaken, since we would depend on the new `TestSubject` containing the information  on which component was changed from the original **control** `TestSubject` at the time of its construction (the `component` label in current implementation).